apache/trafficcontrol
# Add change log count to user response payload and username as query param in logs API

## What does this PR (Pull Request) do?


- [x] This PR fixes #5451 


## Which Traffic Control components are affected by this PR?


- Documentation
- Traffic Ops
- Traffic Portal
-  CDN in a Box

## What is the best way to verify this PR?

1. Use GET curl calls for **log** api for any permutation of limit, days, and username
```
https://<TO-API host:port>/api/4.0/logs?days=10&limit=3
OR 
https://<TO-API host:port>/api/4.0/logs?username=admin&days=10
```

2. Login to TP, and ensure that change log section on dashboard and change log widget ![image](https://user-images.githubusercontent.com/22248619/121382398-cd37c080-c903-11eb-805e-da5f133f4d02.png) shows summary of logs

3. Use GET curl call to **user** api (https://<TO-API host:port>/api/4.0/users), and one should see a new field **chaneLogCount** associated with each user.
![image](https://user-images.githubusercontent.com/22248619/121382735-21db3b80-c904-11eb-9399-0671482d49f1.png)

4. Login to TP, go to Users section, click on select table column ![image](https://user-images.githubusercontent.com/22248619/121382430-d4f76500-c903-11eb-8282-aaca8ec4d901.png), and you should see the number of change logs associated with each user.
![image](https://user-images.githubusercontent.com/22248619/121383281-901ffe00-c904-11eb-815e-3f97fbdc2e38.png)

![image](https://user-images.githubusercontent.com/22248619/121383325-9910cf80-c904-11eb-86eb-407756c7a10a.png)

5. Login to TP, go to Users section, and then click on a user. `View Change Log` button should exists. Click the button to see the logs associated with the user (user should be re-routed to change log page)

6. Run API/v4 tests for logs

## If this is a bug fix, what versions of Traffic Control are affected?



## The following criteria are ALL met by this PR




## Additional Information
